scalyr_agent/builtin_monitors/journald_monitor.py

```diff
@@ -562,12 +556,22 @@ def _process_entries(self):
                 )
 
     def format_msg(
-        self, metric_name, metric_value, extra_fields=None,
+        self, metric_name, metric_value, log_config, extra_fields=None,
     ):
         string_buffer = six.StringIO()
 
-        if self._format_version == 2:
-            string_buffer.write('%s "%s"' % (metric_name, metric_value))
+        detect_escaped_strings = False
+        emit_raw_details = False
+        if log_config:
+            detect_escaped_strings = log_config.get("detect_escaped_strings", False)
+            emit_raw_details = log_config.get("emit_raw_details", False)
+
+        if emit_raw_details or (
```

Comment:
Let's remove some duplicate code here and below.  Let's have a method called `format_key_value(format, key, value, emit_raw, detect_ecaped_strings)`.  It returns a string.

For the details, we would invoke `format_key_value('%s %s', 'details', details_value, emit_raw, detect_escaped_strings)`.  For all other, we would just use the format string of `%s=%s`.  Does that make sense?  It would eliminate having to have two places with those double quotes.

Reply:
For the extra fields I think we discussed not taking into account `emit_raw_details`, should this still be the case you think?

Reply:
Yes, sorry, meant to explicitly say that in my comment.  You coded it up right.. always passing in `False` there.